commons/src/main/java/io/aiven/kafka/connect/common/source/task/PartitionInFilenameDistributionStrategy.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.source.task;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.config.ConfigException;
+
+import org.codehaus.plexus.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PartitionInFilenameDistributionStrategy will determine which files to process for each task, based on the partition
+ * number defined in the filename e.g. the default object names created by the sink connector ex.
+ * topicname-{{partition}}-{{start_offset}} the partition can be extracted to have one task running per partition.
+ *
+ */
+public final class PartitionInFilenameDistributionStrategy implements SourceObjectDistributionStrategy {
+    private final static Logger LOG = LoggerFactory.getLogger(PartitionInFilenameDistributionStrategy.class);
+    private final static String NUMBER_REGEX_PATTERN = "(\\d)+";
+    // Use a named group to return the partition in a complex string to always get the correct information for the
+    // partition number.
+    private final static String PARTITION_NAMED_GROUP_REGEX_PATTERN = "(?<partition>\\d)+";
+    private final static String PARTITION_PATTERN = "\\{\\{partition}}";
+    private final static String START_OFFSET_PATTERN = "\\{\\{start_offset}}";
+    private final static String TIMESTAMP_PATTERN = "\\{\\{timestamp}}";
+    public static final String PARTITION = "partition";
+    private Pattern partitionPattern;
+
+    private int maxTasks;
+
+    PartitionInFilenameDistributionStrategy(final int maxTasks, final String expectedSourceNameFormat) {
+        configureDistributionStrategy(maxTasks, expectedSourceNameFormat);
+    }
+
+    /**
+     *
+     * @param sourceNameToBeEvaluated
+     *            is the filename/table name of the source for the connector.
+     * @return Predicate to confirm if the given source name matches
+     */
+    @Override
+    public boolean isPartOfTask(final int taskId, final String sourceNameToBeEvaluated) {
+        final Matcher match = partitionPattern.matcher(sourceNameToBeEvaluated);
+        if (match.find()) {
+            return toBeProcessedByThisTask(taskId, maxTasks, Integer.parseInt(match.group(PARTITION)));
```

Comment:
there could be a Number format exception here

Reply:
I think this is ok, because the regex is specifically looking for a decimal and as long as match.find() returns true it should always find the decimal for parsing.